pkg/azuredx/datasource.go

```diff
@@ -78,8 +84,13 @@ func (adx *AzureDataExplorer) QueryData(ctx context.Context, req *backend.QueryD
 	backend.Logger.Debug("Query", "datasource", req.PluginContext.DataSourceInstanceSettings.Name)
 	res := backend.NewQueryDataResponse()
 
+	authorization, err := adx.serviceCredentials.QueryDataAuthorization(req)
```

Comment:
for readability, I would check the onBehalfOf property here:

```
authHeader := ""
if adx.settings.EnableOnBehalfOf {
  authHeader = adx.GetOnBehalfOfBearerToken(...
}

Reply:
It's all about encapsulation. Why pollute the query method with authorization logic?

Reply:
If I just read `authorization, err := adx.serviceCredentials.QueryDataAuthorization(req)` that would mean for me that we are somehow managing all authorization in that method, when really what it's doing is retrieving an optional header (falling back to the default auth method if not configured).

In any case, just naming again.

Reply:
You are right. I intend to do "manage all authorization in that method" in a follow up commit, waiting for this to happen first. I can push the PR already for you to see.

Reply:
Since @sunker also felt on this issue I added the followup in here (to see and discuss).